include/sharg/auxiliary.hpp

```diff
@@ -109,6 +109,10 @@ struct parser_meta_data // holds all meta information
      *        "./my_read_mapper -s 3 --my_flag path/infile1").
      */
     std::vector<std::string> examples;
+
+    /*!\brief The executable name, with additional subcommands
+     */
+    std::vector<std::string> executable_name;
```

Comment:
This needs to provide more information for the user since it is public. Why should I specify the executable name?

Reply:
From what I gathered, it doesn't have to be public right now. Could be private, or even handled differently altogether. @SGSSGene correct me if I am wrong :D

Reply:
I'd either make it private, as in literally private (parser can befriend meta), or give an explanation why the user might want to change this instead of defaulting it to argv[0]

Reply:
Mhm, `\private` kind of hides it from the user.

If it were private, we would have to:
* Theoretically, make it a class, with rule of six
* Add some test accessor such that we can actually access the `executable_name` in the tests

I'm uncertain if there could be use cases where you actually want to customize the `executable_name`, @SGSSGene would have to enlighten us :)

I agree that `private:` would be technically more correct, if it is not supposed to be accessible to the user at all. However, it also adds some overhead that might not be worth it?

Reply:
Meeting Notes:
We decided another solution where `executable_name` is part of the parser instead of the meta object. This doesn't change the API at all.